containers/daap-python-base/src/var/task/data_platform_logging.py

```diff
@@ -140,35 +167,36 @@ def error(self, msg, *args, **kwargs):
 
         """
         log_dict = _make_log_dict("error", msg, self.extra, inspect.stack()[1].function)
-        self.log_list_dict.append(log_dict)
+        self._write_log_dict_to_s3_json(log_dict)
         self.logger.error(msg, *args, extra=self.extra, **kwargs)
 
-    def write_log_dict_to_s3_json(self, bucket: str, **kwargs):
+    def _write_log_dict_to_s3_json(
+        self, log_line: dict, additional_args: dict = s3_security_opts
+    ) -> None:
         """
         writes the list of log dicts to s3 as a json file in the correct
         format for an athena table
         """
         s3_client = boto3.client("s3")
-        date = datetime.now().strftime("%Y-%m-%d")
-        date_time = datetime.now().strftime("%Y-%m-%dT%H:%M:%S:%f")[:-3]
-        if (
-            "lambda_name" in self.extra.keys()
-            and "data_product_name" in self.extra.keys()
-        ):
-            key = os.path.join(
-                "logs",
-                "json",
-                f"lambda_name={self.extra['lambda_name']}",
-                f"data_product_name={self.extra['data_product_name']}",
-                f"date={date}",
-                f"{date_time}_log.json",
+
+        try:
+            response = s3_client.get_object(
+                Bucket=self.log_bucket_path.bucket, Key=self.log_bucket_path.key
             )
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "NoSuchKey":
+                jlines = ""
+            else:
+                raise e
```

Comment:
I think we should completely suppress exceptions in this function, because it could cause the data ingestion to fail completely in a situation where it would otherwise continue working.

If there is something wrong with logging to s3, we can catch that separately by monitoring the bucket itself. We should be able to set up cloudwatch alerts to ping us if nothing has been written to the bucket in some time window.

Also I think we will still be able to access the default stdout logging if the s3 logging fails(?)

Reply:
This is true, should exit the method if unexpected error though to avoid creating a 1 line log file for the last log call. Yeah stdout should still be available if this method was to fail.